protocol/v1/qbft/msgqueue/queue.go

```diff
@@ -187,14 +187,24 @@ func (q *queue) Peek(n int, idx Index) []*spectypes.SSVMessage {
 
 // WithIterator looping through all indexes and return true when relevant and pop
 func (q *queue) WithIterator(n int, peek bool, iterator func(index Index) bool) []*spectypes.SSVMessage {
+	indices := make([]Index, len(q.items))
+	i := 0
```

Comment:
Does `i` not get incremented anywhere? I'd suggest `append`ing to `indices` in the `for`-loop, it would be less error-prone.